### PR TITLE
Refactor and add helpers

### PR DESCRIPTION
**Pull request by Sgalvin36**
Refactored classes Teams, Games, Game_Teams to accept CSV row.

Refactor season_stats to combine several methods and simplify reading

Refactor league_stats to combine several methods and all stats are now manager methods calling helper methods

Organized all specs to pass and coverage at 100%

Spec_harness passes all methods tested.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -31,54 +31,23 @@ def self.from_csv(info)
...
-            :type => row[:type],
-            :date_time => row[:date_time],
-            :away_team_id => row[:away_team_id],
-            :home_team_id => row[:home_team_id],
-            :away_goals => row[:away_goals],
-            :home_goals => row[:home_goals],
-            :venue => row[:venue],
-            :venue_link => row[:venue_link]}
-            games << Game.new(class_info)
+            games << Game.new(row)
```
Comment (CCirbo) [inline lib/stat_tracker.rb:34]: This is looking so clean and fantastic. Refactoring like this is clean and more dynamic and very DRY

**Comment by stefanjshepard** [comment]
All that's left is minimal spacing, tabs and code cleanup.  Excellent!

File: lib/league_stat.rb [hunk 2]
```diff
@@ -1,94 +1,64 @@
...
-        end
+        get_team_name(home_away_goals_and_games("away", "lowest"))
     end
 
     def lowest_scoring_home_team
-        @teams.each do |team|
-            return team.team_name if team.team_id == team_goals_and_games("home", "lowest")
+        get_team_name(home_away_goals_and_games("home", "lowest"))
+    end
+
```
Comment (CCirbo) [inline lib/league_stat.rb:35]: Great refactoring, easier to read and great way to reuse code for more methods.

**Review by CCirbo — APPROVED** [review]
Great refactoring

**Pull request merged**